stardog/connection.py

```diff
@@ -129,6 +129,8 @@ def add(self, content, graph_uri=None):
         self._assert_in_transaction()
 
         with content.data() as data:
+            if isinstance(data, str):
```

Comment:
considering the latest discussion in slack, this is not intended as a "type check" (although it is). this is because "data" is meant to be a [Content](https://github.com/stardog-union/pystardog/blob/1b070ab35637807c7d657bc6bb83de1bc32129f5/stardog/content.py#L11).
I am not too sure if this is the way to go. should we accept only `Content`? (if yes, then in this case it's valid to check type and fail for string) should we accept `str` as well, and handle it here?

Reply:
I don't think we should allow this to be two different types here when there is already a `Raw` content subclass that can and should be honoring the `content_encoding` attribute. 

```
    # set content_encoding to `utf-8` by default
    @contextlib.contextmanager
    def data(self):
        yield self.raw.encode(self.content_encoding)
```

Reply:
After discussing this with @cullan we came to the conclusion that the encoding should be done by the end user in this case.  Raw was originally intended to manage strings of triples, but since it uses the 'data' context manager, in reality nothing stops you to open a file ( for example `Raw(open('filename'))` ) or pass an http requests ( `Raw(requests.get(self.url, stream=True)` )) as the first argument to Raw(). In this case, it seems we can even pass a mysql / mongodb or anything at all basically, hence we can't really default `.encode('self.content_encoding')`. to utf-8. I added some documentation and some tests instead, in the case we want to send utf-8 characters.